### DISCUSSION TITLE
Better password manager support for mobile

### DISCUSSION THREAD

**Discussion by Non-Binary-Programmer** [ppy/osu#22565] [Q&A]
I don't know if there's an easy way to do this, but it would be great if there was a way to get the system password manager prompt to show up when trying to log in. If there isn't an easy way to do this, then it would be great to get a paste button for the password section, at least, because most people aren't familiar with the paste gesture (and I'm not sure if there even is one for android), and it's annoying to fit all three fingers into the text bar on smaller devices.

**Comment by peppy** [comment]
Can agree with both of these suggestions. A paste button in textboxes may be a good stop-gap solution (if we can make it only show on touch devices).

**Reply by jlacroix10** [reply]
I believe in android you can long press to paste into most textboxes. That being said I haven't tried with osu.

Password manager support may be keyboard dependent as well, I use SwiftKey, and bitwarden integrates pretty well with most apps can't say much about other combinations.

I'm curious to take a look at it though, I will do that later when I have time.

**Reply by peppy** [reply]
Long press won't work with osu!. Our text boxes are completely custom.

**Comment by Susko3** [comment]
Pasting will be supported with https://github.com/ppy/osu/pull/24184. Integrating with the system password manager prompt is considerably harder, requiring changes across many osu!framework components.

**Reply by hburn7** [reply]
Considering that PR is now closed, are there any planned/ongoing efforts to add copy/paste functionality to mobile?
